examples/basics/multi_input_example.exs

```diff
@@ -17,11 +17,12 @@ defmodule XOR do
     |> Axon.dense(1, activation: :sigmoid)
   end
 
-  defp batch do
-    x1 = Nx.tensor(for _ <- 1..32, do: [Enum.random(0..1)])
-    x2 = Nx.tensor(for _ <- 1..32, do: [Enum.random(0..1)])
+  defp batch(key) do
+    {x1, next_key} = Nx.Random.randint(key, 0, 1 + 1, shape: {32, 1})
+    {x2, next_key} = Nx.Random.randint(next_key, 0, 1 + 1, shape: {32, 1})
```

Comment:
_Normally_, I prefer to avoid variable name rebinding in Elixir. However, I like keeping the same name for the random_key variable to ensure that I'm always using the latest state, so I think we should use this in the examples.

```suggestion
    {x1, random_key} = Nx.Random.randint(random_key, 0, 1 + 1, shape: {32, 1})
    {x2, random_key} = Nx.Random.randint(random_key, 0, 1 + 1, shape: {32, 1})
```

Reply:
I originally write 1+1 because it gives you sugestion that the interval in randint is rhs exclusive, simple 2 does not gives you this sugestion. This is a special case where it is not so clear but I believe that for interval [0, 100] if you write randint(0, 100+1) it is much clearer.

Reply:
Even if it were 100+1 I would still be caught up in that for quite some time. It's probably better to add a comment prefacing with exactly the reason why we have 2 (or 101 in the other case).

We can also make the docs for Nx.Random more clear because they're just using math interval notation instead of explaining in English

Reply:
https://github.com/elixir-nx/nx/pull/1263